index.js

```diff
@@ -1,122 +1,110 @@
 const core = require('@actions/core');
 const shell = require('shelljs');
 
-async function bcc() {
-  core.info(`Start to build env`);
-  core.debug((new Date()).toTimeString()); // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
-  core.info(`Linux header`);
-  if (shell.exec("sudo apt-get install -y linux-headers-`uname -r`").code !== 0){
-    shell.echo("fail to install linux headers");
-    shell.exit(1);
-  }
-  let bcc_version = core.getInput('bcc_version');
-  core.debug(bcc_version);
-  if (bcc_version === undefined || bcc_version == null || bcc_version.length === 0) {
-    bcc_version="0.25.0";
-  }
-  core.info(`Get BCC with version:` + bcc_version);
-  if (shell.exec("wget https://github.com/sustainable-computing-io/kepler-ci-artifacts/releases/download/v"+bcc_version+"/bcc_v"+bcc_version+".tar.gz").code !==0){
-    shell.echo("fail to get BCC deb");
-    shell.exit(1);
-  }
-  if (shell.exec("tar -zxvf bcc_v"+bcc_version+".tar.gz").code !== 0) {
-    shell.echo("fail to untar BCC deb");
-    shell.exit(1);
+function getInputOrDefault(inputName, defaultValue) {
+  const input = core.getInput(inputName);
+  if (input === undefined || input == null || input.length === 0) {
+    return defaultValue;
   }
-  if (shell.exec("sudo dpkg -i libbcc_"+bcc_version+"-1_amd64.deb").code !== 0) {
-    shell.echo("fail to install BCC deb");
+  return input;
+}
+
+function executeCommand(command, errorMessage) {
+  const result = shell.exec(command);
+  if (result.code !== 0) {
+    shell.echo(errorMessage);
     shell.exit(1);
   }
-  return
 }
 
-async function libbpf() {
+function installLinuxHeaders() {
   core.info(`Linux header`);
-  if (shell.exec("sudo apt-get install -y linux-headers-`uname -r`").code !== 0){
-    shell.echo("fail to install linux headers");
-    shell.exit(1);
-  }
+  executeCommand("sudo apt-get install -y linux-headers-`uname -r`", "fail to install linux headers");
+}
+
+function installBcc(bcc_version) {
+  core.info(`Get BCC with version:` + bcc_version);
+  executeCommand("wget https://github.com/sustainable-computing-io/kepler-ci-artifacts/releases/download/v"+bcc_version+"/bcc_v"+bcc_version+".tar.gz", "fail to get BCC deb");
+  executeCommand("tar -zxvf bcc_v"+bcc_version+".tar.gz", "fail to untar BCC deb");
+  executeCommand("sudo dpkg -i libbcc_"+bcc_version+"-1_amd64.deb", "fail to install BCC deb");
+}
+
+function installXgboost(artifacts_version, xgboost_version) {
```

Comment:
may I know if Xgboost is available on GHA default runner?

Reply:
yes, xgboost is needed in default runner